src/Microsoft.AspNetCore.Razor.Language/Legacy/Block.cs

```diff
@@ -214,6 +235,12 @@ public override void Accept(ParserVisitor visitor)
             visitor.VisitBlock(this);
         }
 
+        internal void ChildChanged()
+        {
+            // A span that's in our graph has changed. We'll need to recompute our length the next time we're asked for it.
+            _length = -1;
```

Comment:
This doesn't need to propogate up any farther?

Reply:
Nevermind, you do it in the span class. I would have thought this block would have been responsible for notifying it's parent.

Reply:
> responsible for notifying it's parent.

Hmmm, ya gonna go with having the block notify its parent. I like that better 😄